File: Python/7py 2/max_gap.py
"""
|--------------------------------------------------------------------------
| Maximum Gap (Array Series #4)
|--------------------------------------------------------------------------
|
"""

def max_gap(numbers):
   sort = sorted(numbers)
   return max([sort[i]-sort[i-1] for i in range(1, len(sort))])


from numpy import diff
def max_gap(numbers):
   return max(diff(sorted(numbers)))


def max_gap(numbers):
   lst = sorted(numbers)
   return max(b-a for a,b in zip(lst, lst[1:]))


print(max_gap([13,10,2,9,5]),4)
print(max_gap([13,3,5]),8)
print(max_gap([24,299,131,14,26,25]),168)

File: JavaScript/7js 3/maxGap.js
/*
|--------------------------------------------------------------------------
| Maximum Gap (Array Series #4)
|--------------------------------------------------------------------------
|
*/

function maxGap(numbers) {
   let res = [];
   const sort = numbers.sort((a, b) => a - b);
   for (let i = 1; i < sort.length; i++) {
      res.push(sort[i] - sort[i - 1]);
   }
   return Math.max(...res);
}


function maxGap(numbers) {
   return Math.max(
      ...numbers
         .sort((a, b) => a - b)
         .slice(1)
         .map((val, idx) => val - numbers[idx])
   );
}


function maxGap(numbers) {
   return numbers
      .sort((a, b) => a - b)
      .reduce((gap, curr, ind, arr) => (gap < arr[ind + 1] - curr ? arr[ind + 1] - curr : gap), 0);
}


console.log(maxGap([13, 10, 2, 9, 5]), 4);
console.log(maxGap([13, 3, 5]), 8);
console.log(maxGap([24, 299, 131, 14, 26, 25]), 168);